File: qwertycvb/42587.py
def solution(priorities, location):
    print_queue = []
    priority_queue = sorted(priorities, reverse=True)

    for i, p in enumerate(priorities):
        print_queue.append([i, p])

    priority = 1
    while len(print_queue) != 0:
        if print_queue[0][1] < priority_queue[0]:
            print_queue.append(print_queue.pop(0))
        else:
            if print_queue.pop(0)[0] == location:
                return priority
            priority_queue.pop(0)
            priority += 1

File: sudhdkso/42587.cpp
#include <string>
#include <vector>
#include<queue>
#include<iostream>
#include<algorithm>
using namespace std;

int solution(vector<int> priorities, int location) {
	int answer = 0, index = location;
	queue<int> q;
	vector<int> v = priorities;
	for (int i = 0; i < priorities.size(); i++)
		q.push(priorities[i]);

    //복사한 순서를 내림차순으로 정렬
	sort(v.begin(), v.end(), greater<>());

	while (q.size() > 0) { 
        //priorities에서 제일 큰 값과 큐의 제일 앞을 비교 
		if (q.front() < v[0]) {
			q.push(q.front());
			q.pop();
			if (index <= 0)
				index = q.size() - 1;
			else
				index--;

		}
		else {
			v.erase(v.begin());
			q.pop();
			index--;
			if (index < 0) {
				return priorities.size() - q.size();
			}


		}

	}

	return answer;
}